app/actions/v3/service_instance_create_managed.rb

```diff
@@ -185,6 +189,14 @@ def parse_response(fetch_instance, last_operation)
         }
       end
 
+      def validate_service_instance!(instance)
+        if instance
+          instance_already_exists!(instance.name) if instance.create_succeeded? || instance.create_in_progress? || instance.update_succeeded? ||
+            instance.update_in_progress? || instance.update_failed? || instance.last_operation.nil?
```

Comment:
I think that the check for `last_operation.nil?` is already included in the `ServiceOperationMixin`.

`def create_succeeded?` -> `!last_operation? || ...`

`def last_operation?` -> `!last_operation.nil?`

=> so it's `!!last_operation.nil?` = `last_operation.nil?` :smile:

Reply:
Ehm, what? :D I added a method `last_operation_nil` to the `ServiceOperationMixin`.

Reply:
Sorry, that was confusing...

What I meant was that you don't need this explicit check for `last_operation.nil?`. If the `last_operation` is `nil`, `create_succeeded?` will already return `true`.

The same (i.e. no explicit `nil` check) is done for service bindings, e.g. [here](https://github.com/cloudfoundry/cloud_controller_ng/blob/de07d6efddb933ecdfd710aba64fbd37cfedc1ad/app/actions/service_credential_binding_app_create.rb#L65).

So you can omit this check and also remove the `last_operation_nil` method from `ServiceOperationMixin`.

Reply:
ohh right, changed it accordingly.